src/slack/views/app_home.ts

```diff
@@ -102,14 +102,32 @@ const getAvailableFns = async (selectedTool: SUPPORTED_INTEGRATIONS) => {
 
 export const getToolConnectionView = (
   selectedTool: (typeof INTEGRATIONS)[number]['value'] | string | undefined,
-  mcpConnections: McpConnection[] = []
+  mcpConnections: McpConnection[] = [],
+  slackWorkspace: SlackWorkspace
 ): BlockBuilder[] => {
   const select = Elements.StaticSelect({
     placeholder: 'Select a tool',
     actionId: SLACK_ACTIONS.CONNECT_TOOL
   });
 
-  const integrationOptions = INTEGRATIONS.map((integration) =>
+  // Always calculate connected tool values once and freeze
+  const connectedToolValues: SUPPORTED_INTEGRATIONS[] = [];
+  INTEGRATIONS.forEach((integration) => {
+    const relationKey = integration.relation as keyof SlackWorkspace;
+    if (slackWorkspace?.[relationKey]) {
```

Comment:
Null check not required 
```suggestion
    if (slackWorkspace[relationKey]) {
```